lib/ansible/plugins/inventory/constructed.py

```diff
@@ -29,6 +29,12 @@
             default: false
             type: boolean
             version_added: '2.11'
+        group_vars:
```

Comment:
This works too, but it requires repeating the group name you've theoretically already defined in `groups`. Seems like the whole thing could be made more consistent between `keyed_groups` and `groups` if the group->criteria expression mapping under the existing `groups` key was changed to optionally accept a subdict, like:
```
...
groups:
  dude:
    membership_criteria: '{{ ... }}'
    vars:
      dudevar1: hey there
      ansible_group_priority: 2
```

Reply:
I thought of that, but it would either be backwards incompatible or 'dual nature'  i.e string == condition, dict = new form ... none of the options appeal to me, so i went with the simplest one to implement